gettsim/data/unterhalt.yaml

```diff
@@ -1,51 +1,125 @@
-uhv5:
+uhv6_share:
   name:
-    de: Unterhaltsvorschuss für Kinder unter 5 Jahren. Der Prozentsatz in §2 UVG wird angewendet auf (2 x Kinderfrebeitrag (32 (6) EStG) / 12) - Kindergeld für das 1. Kind
+    de: Unterhaltsvorschuss für Kinder bis 6 Jahre. Anteil am Kinderfreibetrag.
```

Comment:
Not since 2016. Before it was used and you need this share.

Reply:
Please state the parameter you mean with Kinderfreibetrag here.

Reply:
Maybe this would be a case where an "end date" or so for a parameter would be helpful? "obsolete_after"?

Reply:
I don't think that adding end dates is helpful (in general and here). 

You will always have to look at the source code to see which law is in force at a specific date.

Anyway, this share is a parameter of a law that in still in force.

Reply:
Sure, but this share is not.

I agree we should not duplicate things and have the same dates in two places. I also hope that eventually, we can define a bit more in the parameters files than we do currently.

Reply:
> Sure, but this share is not.

Not part of §1612a anymore? Not in force?

The description key of this parameter (`uhv6_share`) says: "§1612a (1) Nr .1 BGB". Looking at https://www.buzer.de/gesetz/6597/a93318.htm it seems to me the law still has this parameter and is in force. With §1612a (4) maybe irrelevant for gettsim. But it is still there.

Btw how does that even work: having §1612a (1) and §1612a (4) at the same time? Seems they lead to different results. Do we just decide to only look at (4) and ignore (1)?

Reply:
> > Sure, but this share is not.
> 
> Not part of §1612a anymore? Not in force?
> 
> The description key of this parameter (`uhv6_share`) says: "§1612a (1) Nr .1 BGB". Looking at https://www.buzer.de/gesetz/6597/a93318.htm it seems to me the law still has this parameter and is in force. With §1612a (4) maybe irrelevant for gettsim. But it is still there.

I meant the reference to Kinderfreibetrag; did not check the source until now. 

But this brings us back to the old "depends on" discussion we had at some point. I guess this would be a case where that would have changed over time? 

> Btw how does that even work: having §1612a (1) and §1612a (4) at the same time? Seems they lead to different results. Do we just decide to only look at (4) and ignore (1)?

Probably yes. https://www.buzer.de/gesetz/11821/al64917-0.htm 

Are the results different?

Reply:
> Are the results different?

yes I think that is e.g. 221 vs 212 euro in the example I mention here: 
https://github.com/iza-institute-of-labor-economics/gettsim/pull/74#pullrequestreview-330520136

Reply:
I am lost now. The link I just sent for (4) defines precise numbers and none is below 335 € ?

Reply:
that is correct. subtract a `kgeld1` to get the Unterhaltsvorschuss.

Reply:
Because you have to subtract Kindergeld (204€ right now) in order to get the proper UHV. 

I don't know if the Kinderfreibetrag Referenz in §1612a BGB does have any purpose now. It used to have until 2016, which is why I did not delete it.

Reply:
Great, you just have to be patient with an old man.

So until end of 2015, `uhv*_share` is relevant and `uhv*_amount` thereafter. Whatever we decide in the end, we should have some cross-reference eventually so one can infer that the parameter is superseded by another one.

Reply:
I would sketch it as follows:
until 2016-12-31 `uhv = kifrei_s * _share - kgeld1` and thereafter `uhv = _amout - kgeld1`. 

Plus, I thin Eric only implemented uhv starting 2017 because before UV was only a temporary benefit?

Reply:
Exactly. But if someone wants to model it pre 2017, the params are there.

Reply:
So what is missing? The time limits?

Reply:
There is no function right now at all pre 2017 doing `kifreb_s * share - kgeld1`.

Reply:
Nothing is really missing. I basically just wanted:
> Please state the parameter you mean with Kinderfreibetrag here.

as we now have `kifrei` and `kifrei_s`.

I did not question the fact that the shares are added but not (yet) used. I'm absolutely fine with that.

Reply:
Can somebody pls open an issue for the correction of that stuff mentioned above and also include renaming these guys? Maybe `ki_freib_gesamt` and `ki_freib_ex_min` ? Not married to those terms, but it should become clear from the name what is meant.

Reply:
See #130.